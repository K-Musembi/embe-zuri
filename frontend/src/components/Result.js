import React from "react";

const Result = ({ result }) => {
    
    return (
        <div className="result-panel">
            {result ? (
                <div>
                    <h2>Prediction outcome</h2>
                    <p>Disease: {result.prediction}</p>
                    <p>Confidence: {result.confidence}%</p>
                </div>
            ) : (
                <div className="no-result">
                    <p>Please upload an image</p>
                </div>
            )}
        </div>
    );
};

export default Result;
