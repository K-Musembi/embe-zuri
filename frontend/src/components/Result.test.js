import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Result from './Result';

// Mock fetch globally
global.fetch = jest.fn(() =>
  Promise.resolve({
    ok: true,
    json: () => Promise.resolve({ data: 'mocked data' }),
  })
);

describe('Result Component', () => {
  const mockResult = {
    prediction: "Healthy",
    confidence: 85,
    remedies: ["Use antifungal spray", "Ensure proper ventilation"],
  };

  it('renders correctly with result', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Result result={mockResult} onPrediction={jest.fn()} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
