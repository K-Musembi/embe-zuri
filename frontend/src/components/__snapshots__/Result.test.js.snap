// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component renders correctly with result 1`] = `
<div
  className="result-panel"
>
  <div
    className="result-container"
  >
    <div
      className="prediction-outcome"
    >
      <h2>
        Prediction outcome
      </h2>
      <h3
        className="healthy"
      >
        Healthy
      </h3>
      <h3>
        Confidence: 
        85
        %
      </h3>
    </div>
    <div
      className="recommendations"
    >
      <h2>
        Recommendations
      </h2>
      <div
        className="remedy"
      >
        Use antifungal spray
      </div>
      <div
        className="remedy"
      >
        Ensure proper ventilation
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    More Information
  </button>
</div>
`;
